src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/AdapterTest.php

```diff
@@ -65,4 +66,37 @@ public function testLdapQueryIterator()
         $this->assertEquals(array('Fabien Potencier'), $entry->getAttribute('cn'));
         $this->assertEquals(array('fabpot@symfony.com', 'fabien@potencier.com'), $entry->getAttribute('mail'));
     }
+
+    public function testLdapQueryScopeBase()
+    {
+        $ldap = new Adapter($this->getLdapConfig());
+
+        $ldap->getConnection()->bind('cn=admin,dc=symfony,dc=com', 'symfony');
+
+        $query = $ldap->createQuery('cn=Fabien Potencier,dc=symfony,dc=com', '(objectclass=*)', array(
+           'scope' => Query::SCOPE_BASE,
+        ));
+        $result = $query->execute();
+
+        $entry = $result[0];
+        $this->assertEquals($result->count(), 1);
+        $this->assertEquals(array('Fabien Potencier'), $entry->getAttribute('cn'));
+    }
+
+    public function testLdapQueryScopeOneLevel()
+    {
+        $ldap = new Adapter($this->getLdapConfig());
+
+        $ldap->getConnection()->bind('cn=admin,dc=symfony,dc=com', 'symfony');
+
+        $one_level = $ldap->createQuery('ou=Components,dc=symfony,dc=com', '(objectclass=*)', array(
+            'scope' => Query::SCOPE_ONELEVEL,
+        ))->execute()->count();
+
+        $subtree = $ldap->createQuery('ou=Components,dc=symfony,dc=com', '(objectclass=*)', array(
+            'scope' => Query::SCOPE_SUBTREE,
+        ))->execute()->count();
+
+        $this->assertNotEquals($one_level, $subtree);
```

Comment:
I do not feel that this tests anything useful (Apart from checking that a one level scope is different from a subtree). We should actually check whether we have the expected content in the output from the test.